### PR TITLE
Add option to clear panel buffer

### LINKED ISSUE #1247: Add the ability to remove the panel buffer
Taken from #1226

### PR DESCRIPTION
**Pull request by bnmajor**
Fixes #1247

### CODE REVIEW

File: hexrd/ui/calibration/panel_buffer_dialog.py [hunk 1]
```diff
@@ -160,3 +161,18 @@ def mode(self, v):
     def update_mode_tab(self):
         mode_tab = getattr(self.ui, self.mode + '_tab')
         self.ui.tab_widget.setCurrentWidget(mode_tab)
+
+    def clear_panel_buffer(self):
+        # Clear the config options on the internal config
+        config = HexrdConfig().config
+        detector_config = config['instrument']['detectors'][self.detector]
```
Comment (psavery) [inline hexrd/ui/calibration/panel_buffer_dialog.py:168]:
```suggestion
        detector_config = HexrdConfig().detector(self.detector)
```

File: hexrd/ui/calibration/panel_buffer_dialog.py [hunk 2]
```diff
@@ -160,3 +161,18 @@ def mode(self, v):
...
+
+        buffer_default = {'status': 0}
+        buffer = detector_config.setdefault('buffer', buffer_default)
+
+        # Must match the detector size
+        detector_shape = (detector_config['pixels']['rows']['value'],
+                          detector_config['pixels']['columns']['value'])
+
+        empty_mask = np.ones(detector_shape, dtype=bool)
+        buffer['value'] = empty_mask
```
Comment (psavery) [inline hexrd/ui/calibration/panel_buffer_dialog.py:178]: I believe that we can just set `buffer['value'] = [0., 0.]` to nullify it, or even `buffer['value'] = None` (but we would need to make sure we are handling the `None` case properly in the GUI - I think we are not, even though it is valid in hexrd).

File: hexrd/ui/calibration/panel_buffer_dialog.py (hunk 1 above)
Comment (bnmajor) [reply hexrd/ui/calibration/panel_buffer_dialog.py]: Perfect, fixed.

File: hexrd/ui/calibration/panel_buffer_dialog.py (hunk 2 above)
Comment (bnmajor) [reply hexrd/ui/calibration/panel_buffer_dialog.py]: Much cleaner, thank you. Fixed to use the `buffer['value'] = [0., 0.]` option.

File: hexrd/ui/calibration/panel_buffer_dialog.py [hunk 3]
```diff
@@ -160,3 +161,12 @@ def mode(self, v):
...
         self.ui.tab_widget.setCurrentWidget(mode_tab)
+
+    def clear_panel_buffer(self):
+        # Clear the config options on the internal config
+        detector_config = HexrdConfig().detector(self.detector)
+
+        buffer_default = {'status': 0}
+        buffer = detector_config.setdefault('buffer', buffer_default)
+
+        buffer['value'] = buffer['value'] = [0., 0.]
```
Comment (psavery) [inline hexrd/ui/calibration/panel_buffer_dialog.py:172]: Uh oh, looks like `buffer['value'] =` is there twice?

**Review by psavery — APPROVED** [review]
Once my last comment is addressed, then LGTM

File: hexrd/ui/calibration/panel_buffer_dialog.py (hunk 3 above)
Comment (bnmajor) [reply hexrd/ui/calibration/panel_buffer_dialog.py:172]: Fixed!

**Pull request merged**